### ISSUE TITLE
[Enhancement] Extending "Disconnect Pump"

### ISSUE THREAD

**Issue by triplem** [MilosKozak/AndroidAPS#1191]
If I have disconnected my pump for eg 1hour and would like to extend the timeframe, I need to Reconnect and then need to disconnect again. I would like to have an opportunity to "Extend Disconnection by x hours".

**Comment by triplem** [comment]
The same is true for "Suspend loop".

As an alternative we could ask the user for a timeframe and/or for a concrete time, when the pump is going to get connected again. If the pump is already disconnected, the user could the provide a different time.

**Comment by jotomo** [comment]
The menu as is is rather long. Take a look at the TT menu, maybe we can use the same idea here: there are preconfigured defaults that can be quickly set and a 'custom' option which options a dialog which allows to set custom times. If a TT is active there's a 'cancel' item. If  suspended/disconnected there could be an 'extend' option?

**Comment by AdrianLxM** [comment]
Once you are in "suspended" or "disconnected" state, the menu is rather small.

**Comment by jotomo** [comment]
Yup, but it sounded like @triplem was thinking about a dialog to input the extension time into, so I thought we might use one dialog to input suspend/disconnect/extension time. Just an idea (for 2.1) :-)

**Comment by triplem** [comment]
Well, my first thought was just to have an additional menu entry while in "suspend" or "disconnect" mode with "Extending Suspend by x  hours". Basically the same like already in the menu if in Cloed Loop, but by just Extending the already set end time. 

Inputting an an own start and/or end time would be an additional think, which came to my mind while writing this issue.

And yes, this is 2.x, that's why I have it marked as enhancement.
